### PR TITLE
ROMIO: Always build subroutines with prefix MPIOI_File_

### PR DESCRIPTION
**Pull request by wkliao**
## Pull Request Description
* This PR is for preparing ROMIO to be built stand alone
* Subroutines with prefix MPIOI_File_ are used no matter
  MPIO_BUILD_PROFILING is defined or not

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

File: src/mpi/romio/mpi-io/read.c [hunk 1]
```diff
@@ -60,8 +60,6 @@ int MPI_File_read(MPI_File fh, void *buf, int count, MPI_Datatype datatype, MPI_
     return error_code;
 }
 
-/* prevent multiple definitions of this routine */
-#ifdef MPIO_BUILD_PROFILING
```
Comment (hzhou) [inline src/mpi/romio/mpi-io/read.c:64]: On platforms that doesn't support weak symbols, we will build the code twice, once with every functions included but with top `MPI_xxx` functions defined as `PMPI_xxx`; and a second time with all the `MPI_xxx` functions but stripping all the internal function. Normally, both libraries are linked. When user use the profiling interface, they will replace the latter with their own profiling library.

If you remove this macro, the internal functions will be included in both libraries and causing duplication when linked together.

I am commenting based on how profiling works for core MPICH. ROMIO behaves similarly, right?

Comment (hzhou) [reply src/mpi/romio/mpi-io/read.c:64]: Mac doesn't support weak symbols, so it can be used for testing.

Comment (wkliao) [reply src/mpi/romio/mpi-io/read.c:64]: I tested this by adding '--disable-weak-symbols' at configure command line and without it.
Please let me know the case that failed this PR.

Comment (wkliao) [reply src/mpi/romio/mpi-io/read.c:64]: Does MPICH CI include building on a Mac?

Comment (raffenet) [reply src/mpi/romio/mpi-io/read.c:64]: Yes, we have macOS CI builds.

**Comment by raffenet** [comment]
test:mpich/ch3/most

**Comment by wkliao** [comment]
Close as this affects only #6956

**Pull request closed without merging**